core/src/services/s3/core.rs

```diff
@@ -468,6 +469,11 @@ impl S3Core {
         if let Some(limit) = limit {
             write!(url, "&max-keys={limit}").expect("write into string must succeed");
         }
+        if let Some(start_after) = start_after {
+            let start_after = format!("{}{}", &p, &start_after);
```

Comment:
Please use `build_abs_path` to build it. `start_after` should be relative to `root` instead of `path`.